codegen/templates/templates.go

```diff
@@ -517,7 +517,14 @@ func wordWalker(str string, f func(*wordInfo)) {
 		}
 
 		matchCommonInitial := false
-		if commonInitialisms[strings.ToUpper(word)] {
+		upperWord := strings.ToUpper(word)
+		if commonInitialisms[upperWord] {
+			switch upperWord {
```

Comment:
```suggestion
// If the uppercase word (string(runes[w:i]) is "ID" or "IP"
// AND
// the word is the first two characters of the str
// AND  
// that is not the end of the word
// AND 
// the length of the string is greater than 3
// AND
// the third rune is an uppercase one
// THEN
// do NOT count this as an initialism.
			switch upperWord {
```

Reply:
Thank you for your suggestion!
I applied it and pushed.